zipkin-lens/src/components/TracePage/TraceSummaryHeader.jsx

```diff
@@ -32,84 +35,135 @@ const defaultProps = {
 };
 
 const useStyles = makeStyles(theme => ({
-  headerTop: {
-    borderColor: theme.palette.grey[300],
+  root: {
+    paddingLeft: theme.spacing(3),
+    paddingRight: theme.spacing(3),
+  },
+  upperBox: {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'space-between',
+    borderBottom: `1px solid ${theme.palette.grey[300]}`,
+  },
+  serviceNameAndSpanName: {
+    display: 'flex',
+    alignItems: 'center',
   },
   serviceName: {
     textTransform: 'uppercase',
   },
   spanName: {
-    color: theme.palette.text.hint,
+    color: theme.palette.text.secondary,
+  },
+  jsonUploaderAndSearchInput: {
+    display: 'flex',
+    alignItems: 'center',
+    paddingRight: theme.spacing(4),
+  },
+  lowerBox: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginTop: theme.spacing(0.5),
+    marginBottom: theme.spacing(0.5),
+  },
+  traceInfo: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  traceInfoEntry: {
+    marginRight: theme.spacing(1),
+    display: 'flex',
+  },
+  traceInfoLabel: {
+    fontWeight: 'bold',
+    color: theme.palette.grey[600],
+  },
+  traceInfoValue: {
+    fontWeight: 'bold',
+    marginLeft: theme.spacing(0.8),
+  },
+  saveButton: {
+    fontSize: '0.5rem',
+    lineHeight: 0.8,
+  },
+  saveButtonIcon: {
+    marginRight: theme.spacing(1),
   },
 }));
 
 const TraceSummaryHeader = React.memo(({ traceSummary, rootSpanIndex }) => {
   const classes = useStyles();
 
+  const handleSaveButtonClick = useCallback(() => {
+    if (!traceSummary || !traceSummary.traceId) {
+      return;
+    }
+    fetch(`${api.TRACE}/${traceSummary.traceId}`)
```

Comment:
I think it's fine to `fetch`, but don't we already have the trace's JSON in memory?

Reply:
Strictly speaking, no.
Raw trace data will be converted in the action creator.

https://github.com/openzipkin/zipkin/blob/7e80e71bf682f74d900f866bc9ab1608ac01f6c8/zipkin-lens/src/actions/trace-action.js#L54-L55

And only the converted data are saved as the redux state.

Reply:
Conversion processes of the trace data often takes time, so I thought it should be done asynchronously within Action creator, and I did so.

Reply:
Of course, raw data can also be stored as a redux state.
However, in Trace page, lens currently re-uses data which is fetched in Search page.

https://github.com/openzipkin/zipkin/blob/7e80e71bf682f74d900f866bc9ab1608ac01f6c8/zipkin-lens/src/actions/trace-action.js#L43-L46

Because of this process, I thought that storing raw data in the redux state would make code more complex than necessary.
I thought the simplicity of the implementation is the most important, so I decided to fetch the JSON again when the button was pressed.

Reply:
Sure it makes sense since exporting traces is uncommon anyways